src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -245,8 +245,7 @@ internal unsafe int IndexOfCore(string source, string target, int startIndex, in
             Debug.Assert((options & CompareOptions.OrdinalIgnoreCase) == 0);
             Debug.Assert((options & CompareOptions.Ordinal) == 0);
 
-#if CORECLR
-            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options) && source.IsFastSort() && target.IsFastSort())
+            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options) && source.IsAscii() && target.IsAscii())
```

Comment:
I guess this code can be optimized - we can omit IsAscii and just run IndexOf and figure out if it's not "IsAscii" on the fly.

Reply:
Looks Windows implementation don't have this optimization. we can try just remove the whole optimization and compare the perf with Windows. 

checking IsAscii only here is incorrect in Linux as you can run into issues when having characters like hyphens. 

---
In reply to: [281011011](https://github.com/dotnet/coreclr/pull/24400#discussion_r281011011) [](ancestors = 281011011)